azure-pipelines.yml

```diff
@@ -54,522 +54,568 @@ resources:
+          MAVEN_OPTS: '-Xmx3072m -Dmaven.repo.local=$(MAVEN_CACHE_FOLDER)'
+        pool: .net-bubble-aws-re-team-prod
+        steps:
+          - checkout: self
+            fetchDepth: 1
+
+          - task: Cache@2
+            displayName: Cache Maven local repo
+            inputs:
+              key: maven | its/pom.xml
+              path: $(MAVEN_CACHE_FOLDER)
+
+          - task: DownloadSecureFile@1
+            displayName: 'Download maven settings'
+            name: mavenSettings
+            inputs:
+              secureFile: 'maven-settings.xml'
+
+          - task: DownloadPipelineArtifact@2
+            inputs:
+              buildType: 'current'
+              targetPath: '$(Build.SourcesDirectory)\build'
+              artifactName: build
+
+          - task: NuGetToolInstaller@1
+            inputs:
+              versionSpec: '5.8.0'
+
+          - powershell: |
+              $projectVersion = Get-Content "$(Build.SourcesDirectory)\build\version.txt"
+              Write-Host "##vso[task.setvariable variable=SONAR_PROJECT_VERSION]$projectVersion"
+            displayName: "Get version from artifact file"
+
+          - task: Maven@3
+            displayName: 'Run Maven ITs for $(PRODUCT) $(SQ_VERSION)'
+            env:
+              ARTIFACTORY_USER: $(ARTIFACTORY_PRIVATE_READER_USERNAME)
+              ARTIFACTORY_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
+              ARTIFACTORY_QA_READER_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
+              # For Orchestrator (https://github.com/SonarSource/orchestrator/commit/d5396c75ab77e6088afe58e61b0cd0693ac885f0)
+              ARTIFACTORY_ACCESS_TOKEN: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
+              GITHUB_TOKEN: $(GITHUB_TOKEN)
+              MAVEN_LOCAL_REPOSITORY: $(MAVEN_CACHE_FOLDER)
+              NUGET_PATH: $(NUGETEXETOOLPATH)
+            inputs:
+              goals: 'verify'
+              options: --settings $(mavenSettings.secureFilePath) -B -e -Denable-repo=qa -DtestInclude=$(TEST_INCLUDE) -Dsonar.cfamilyplugin.version=$(SONAR_CFAMILYPLUGIN_VERSION) -Dsonar.csharpplugin.version=$(SONAR_DOTNET_VERSION) -Dsonar.vbnetplugin.version=$(SONAR_DOTNET_VERSION) -Dsonar.runtimeVersion=$(SQ_VERSION) -DscannerForMSBuild.version=$(SONAR_PROJECT_VERSION).$(Build.BuildId) -Dmsbuild.path="$(MSBUILD_PATH)" -Dmsbuild.platformtoolset=$(PLATFORMTOOLSET) -Dmsbuild.windowssdk=$(WINDOWSSDKTARGET)
+              publishJUnitResults: true
+              mavenPomFile: 'its/pom.xml'
+              testResultsFiles: '**/surefire-reports/TEST-*.xml'
+              testRunTitle: 'ITs $(SQ_VERSION)'
+              javaHomeOption: 'JDKVersion'
+              jdkVersionOption: $(JDKVERSION)
+              mavenOptions: $(MAVEN_OPTS)
+              mavenVersionOption: 'Default'
+
+          - bash: git checkout .
+            name: revertPomChanges
+            displayName: Revert changes made to pom.xml to not break cache feature
+
+  - stage: Artifacts
```

Comment:
This is a copy of the [promoteRepox job from promote-stage.yml](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_git/pipelines-yaml-templates?path=/promote-stage.yml&version=GBmaster&line=20&lineEnd=20&lineStartColumn=7&lineEndColumn=25&lineStyle=plain&_a=contents) without the [Burgr report](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_git/pipelines-yaml-templates?path=/promote-stage.yml&version=GBmaster&line=67&lineEnd=118&lineStartColumn=1&lineEndColumn=107&lineStyle=plain&_a=contents).